src/script/index.js

```diff
@@ -24,10 +26,46 @@ for (var i = 0; i < paramPairsLength; i++) {
 delete i
 delete paramPair
 
+function instances_loading_error() {
+    console.error('Failed to fetch servers list from joinmastodon.');
+}
+
+function instances_loaded() {
+    if (this.status !== 200) {
+        instances_loading_error();
+        return;
+    }
+
+    const servers = JSON.parse(this.responseText);
+
+    const chosen_instance = instance.value;
+    const domains = servers.map(obj => obj.domain);
+    if (domains.indexOf(chosen_instance) === -1) {
+        domains.push(chosen_instance);
+    }
```

Comment:
Instance value can be empty — we don't want to populate datalist with it in this case

```suggestion
    if (chosen_instance && domains.indexOf(chosen_instance) === -1) {
        domains.push(chosen_instance);
    }
```

Reply:
Nice catch